cluster/ClusterDatabaseManager.java

```diff
@@ -100,4 +97,26 @@ public void create(String name) {
     Map<String, CoreDatabaseManager> databaseMgrs() {
         return databaseMgrs;
     }
+
+    private <RESULT> RESULT failsafeTask(String name, BiFunction<GraknStub.Cluster, CoreDatabaseManager, RESULT> task) {
+        FailsafeTask<RESULT> failsafeTask = new FailsafeTask<RESULT>(client, name) {
+
+            @Override
+            RESULT run(ClusterDatabase.Replica replica) {
+                return task.apply(client.stub(replica.address()), client.coreClient(replica.address()).databases());
+            }
+
+            @Override
+            RESULT rerun(ClusterDatabase.Replica replica) {
+                return run(replica);
+            }
+        };
+        try {
```

Comment:
`failsafeTask` performs `task` to any replica, and if that replica isn't the primary replica, will retry to the primary replica.

This is unintuitive and will be fixed in the upcoming PRs. The fix will require some changes on the server side.